src/app/plan/components/Wizard/HooksStep.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  AddEditMode,
+} from '../../../common/add_edit_state';
+import HooksFormContainer from './HooksFormContainer';
+
+const classNames = require('classnames');
+const styles = require('./HooksStep.module')
+
+const HooksStep = (props) => {
+  const {
+    updateHookRequest,
+    addHookRequest,
+    isFetchingHookList,
+    migHookList,
+    fetchHooksRequest,
+    hookAddEditStatus,
+    currentPlan,
+    removeHookRequest,
+    watchHookAddEditStatus,
+    isAddHooksOpen,
+    setIsAddHooksOpen
+  } = props;
+
+  const [initialHookValues, setInitialHookValues] = useState({});
+
+  useEffect(() => {
+    if (currentPlan) {
+      fetchHooksRequest(currentPlan.spec.hooks);
+
+    }
+  }, []);
+
+  const onAddEditHookSubmit = (hookValues) => {
+    switch (hookAddEditStatus.mode) {
+      case AddEditMode.Edit: {
+        updateHookRequest(hookValues);
+        setIsAddHooksOpen(false)
+        break;
+      }
+      case AddEditMode.Add: {
+        addHookRequest(hookValues);
+        setIsAddHooksOpen(false)
+        break;
+      }
+      default: {
+        console.warn(
+          `onAddEditSubmit, but unknown mode was found: ${hookAddEditStatus.mode}. Ignoring.`);
+      }
+    }
+  };
+
+  const columns = [
+    { title: 'Name' },
+    { title: 'Definition' },
+    { title: 'Run in' },
+    { title: 'Migration step' },
+  ];
+
+  let rows = []
+  let actions = [];
+  if (migHookList.length > 0) {
+    rows = migHookList.map((migHook, id) => {
+      return {
+        cells: [migHook.hookName, migHook.image, migHook.clusterTypeText, migHook.phase],
+      }
+    });
+    actions = [
+      {
+        title: 'Edit',
+        onClick: (event, rowId, rowData, extra) => {
+          const currentHook = migHookList.find((hook) => hook.hookName === rowData.name.title)
+          setInitialHookValues(currentHook)
+          setIsAddHooksOpen(true)
+          watchHookAddEditStatus(rowData.name.title);
+
+        }
+      },
+      {
+        title: 'Delete',
+        onClick: (event, rowId, rowData, extra) => {
+          removeHookRequest(rowData.name.title, rowData["migration-step"].title)
+        }
+      }
+    ]
+
+  }
+  if (rows.length === 0) {
+    rows = [{
+      heightAuto: true,
+      cells: [
+        {
+          props: { colSpan: 8 },
+          title: (
+            <Bullseye>
+              <EmptyState variant={EmptyStateVariant.full}>
+                <EmptyStateIcon icon={PlusCircleIcon} />
+                <Title headingLevel="h4" size="lg">
+                  No hooks have been added to this migration plan.
+                </Title>
+              </EmptyState>
+            </Bullseye >
+          )
+        },
+      ]
+    }]
+  }
+
+  const hooksFormContainerStyles = classNames(spacing.mtSm, styles.formContainerStyles)
+  return (
+    <Grid>
+      <GridItem className={spacing.mtSm}>
+        <TextContent>
+          <Text component={TextVariants.h2}>
+            Hooks
```

Comment:
We can actually remove this Hooks text header and use `<WizardStepContainer>` in the `WizardComponent` instead.